src/app/features/settings/go-pro/services/go-pro-media.service.ts

```diff
@@ -0,0 +1,197 @@
+import { HttpClient } from '@angular/common/http';
+import { Inject, Injectable } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+import '@capacitor-community/http';
+import {
+  Capacitor,
+  FilesystemDirectory,
+  FilesystemPlugin,
+  Plugins,
+} from '@capacitor/core';
+import { isPlatform } from '@ionic/core';
+import { FILESYSTEM_PLUGIN } from '../../../../shared/capacitor-plugins/capacitor-plugins.module';
+import { CaptureService } from '../../../../shared/capture/capture.service';
+import { blobToBase64 } from '../../../../utils/encoding/encoding';
+import { GoProFile } from '../go-pro-media-file';
+const { Http } = Plugins;
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GoProMediaService {
+  private readonly GO_PRO_FILES_ON_DEVICE_STORAGE_KEY =
+    'GO_PRO_FILES_ON_DEVICE_STORAGE_KEY';
+  private readonly directory = isPlatform('ios')
+    ? FilesystemDirectory.Documents
+    : FilesystemDirectory.Data;
+
+  private readonly rootDir = 'ImageStore';
+
+  private readonly goproBaseUrl = 'http://10.5.5.9:8080';
+
+  constructor(
+    @Inject(FILESYSTEM_PLUGIN)
+    private readonly filesystemPlugin: FilesystemPlugin,
+    private readonly sanitizer: DomSanitizer,
+    private readonly captureService: CaptureService,
+    private readonly httpClient: HttpClient
+  ) {}
+
+  // eslint-disable-next-line class-methods-use-this
```

Comment:
For all the lint rule `class-methods-use-this` issues, I would suggest define these functions in a standalone file and export them, instead of defining them as GoProMediaService's class methods. See `src/app/utils/url.ts` for example (it is also a good place to put url-handling functions)

Reply:
https://github.com/numbersprotocol/capture-lite/pull/1205/commits/2f0ce31a555beee9e9188080997e18003593f517 should resolve